src/symbol/collision_index.js

```diff
@@ -267,13 +267,18 @@ class CollisionIndex {
             const blocking = collisionBoxArray.get(thisTileFeatures[i]);
             const sourceLayer = blocking.sourceLayerIndex;
             const featureIndex = blocking.featureIndex;
+            const bucketIndex = blocking.bucketIndex;
 
             // Skip already seen features.
             if (sourceLayerFeatures[sourceLayer] === undefined) {
                 sourceLayerFeatures[sourceLayer] = {};
             }
-            if (sourceLayerFeatures[sourceLayer][featureIndex]) continue;
-
+            if (sourceLayerFeatures[sourceLayer][featureIndex] === undefined) {
+                sourceLayerFeatures[sourceLayer][featureIndex] = {};
+            }
+            if (sourceLayerFeatures[sourceLayer][featureIndex][bucketIndex]) {
+                continue;
+            }
```

Comment:
Since there are generally fewer bucket indices than feature indices, would it be (slightly) beneficial to switch the order to `sourceLayerFeatures[sourceLayer][bucketIndex][featureIndex]`?

Reply:
🤔 maybe? I wouldn't expect it to make too big a difference...